src/main/java/org/example/manager/SlashCommandManager.java

```diff
@@ -12,38 +11,55 @@
 public class SlashCommandManager {
     private static SlashCommandManager INSTANCE;
     @Getter
-    private final List<SlashCommand> slashCommands = new ArrayList<>();
+    private final List<SlashCommand> genericSlashCommands = new ArrayList<>();
+    @Getter
+    private final List<SlashCommand> insultSlashCommands = new ArrayList<>();
+
+    public static SlashCommandManager getInstance() {
+        if (INSTANCE == null) {
+            INSTANCE = new SlashCommandManager();
+            return INSTANCE;
+        }
+        return INSTANCE;
+    }
 
     private SlashCommandManager() {
-        register(new SlashClear());
-        register(new SlashAddInsult(InsultService.getInstance()));
-        register(new SlashPlay());
-        register(new SlashSkip());
-        register(new SlashVolume());
-        register(new SlashStop());
-        register(new SlashNowPlaying());
-        register(new SlashDeleteInsult());
-        register(new SlashInsultTarget());
+        registerGenericCommand(new SlashClear());
+        registerGenericCommand(new SlashPlay());
+        registerGenericCommand(new SlashSkip());
+        registerGenericCommand(new SlashVolume());
+        registerGenericCommand(new SlashStop());
+        registerGenericCommand(new SlashNowPlaying());
+
+        registerInsultCommand(new SlashAddInsult());
+        registerInsultCommand(new SlashDeleteInsult());
+        registerInsultCommand(new SlashInsultTarget());
+        registerInsultCommand(new SlashConsent());
     }
 
-    private void register(SlashCommand command) {
-        slashCommands.add(command);
+    private void registerGenericCommand(SlashCommand command) {
+        genericSlashCommands.add(command);
     }
 
-    public List<CommandData> getSlashCommandsData() {
+    private void registerInsultCommand(SlashCommand command) {
+        insultSlashCommands.add(command);
+    }
+
+    public List<CommandData> getGenericSlashCommandsData() {
         List<CommandData> data = new ArrayList<>();
 
-        for (SlashCommand slashCommand : slashCommands) {
+        for (SlashCommand slashCommand : genericSlashCommands) {
             data.add(slashCommand.getData());
         }
         return data;
```

Comment:
Loop here could be replaced with stream, and is a duplicated with one below.

Reply:
\+ and addressed, replaced with ArrayList::forEach for both cases.

Reply:
I would see it more like this.

```java
    public List<CommandData> getGenericSlashCommandsData() {
        return getCommandData(genericSlashCommands);
    }

    public List<CommandData> getInsultSlashCommandsData() {
        return getCommandData(insultSlashCommands);
    }

    private List<CommandData> getCommandData(List<SlashCommand> slashCommands) {
        return slashCommands.stream()
                .map(SlashCommand::getData)
                .collect(Collectors.toList());
    }
```